centinel/experiments/baseline.py

```diff
@@ -0,0 +1,271 @@
+
+
+class BaselineExperiment(Experiment):
+    name = "baseline"
+    # country-specific, world baseline
+    # this can be overridden by the main thread
+    input_files = ['country', 'world']
+
+    # we do our own tcpdump recording here
+    overrides_tcpdump = True
+
+    def __init__(self, input_files):
+        self.input_files = input_files
+        self.results = []
+
+        # should we do tcpdump?
+        if os.geteuid() != 0:
+            self.record_pcaps = False
+        else:
+            self.record_pcaps = True
+
+        if os.geteuid() != 0:
+            logging.info("Centinel is not running as root, "
+                         "traceroute will be limited to UDP.")
+            self.traceroute_methods = ["udp"]
+        else:
+            # if running as root, TCP and UDP traceroute
+            # doing all 3 methods takes a lot of time
+            # self.traceroute_methods = ["icmp", "udp", "tcp"]
+            self.traceroute_methods = ["tcp", "udp"]
+
+    def run(self):
+        if self.record_pcaps:
+            self.external_results = {}
+
+        for input_file in self.input_files.items():
+            logging.info("Testing input file %s..." % (input_file[0]))
+            self.results.append(self.run_file(input_file))
+
+    def run_file(self, input_file):
+        file_name, file_contents = input_file
+
+        # Initialize the results for this input file.
+        # This can be anything from file name to version
+        # to any useful information.
+        result = {}
+        result["file_name"] = file_name
+
+
+        http_results = {}
+        tls_results = {}
+        dns_results = {}
+        traceroute_results = {}
+        url_metadata_results = {}
+        file_metadata = {}
+        file_comments = []
+
+        # each pcap file is stored in a separate file
+        # designated by a number. the indexes are stored
+        # in the json file and the pcap files are stored
+        # with their indexes as file names.
+        pcap_results = {}
+        pcap_indexes = {}
+        url_index = 0
+        comments = ""
+
+        # we may want to make this threaded and concurrent
+        for line in file_contents:
+            line = line.strip()
+
+            # parse file comments, if it looks like "key : value",
+            # parse it as a key-value pair. otherwise, just
+            # store it as a raw comment.
+            if line[0] == '#':
+                line = line[1:].strip()
+                if len(line.split(':')) > 1:
+                    key, value = line.split(':', 1)
+                    key = key.strip()
+                    value = value.strip()
+                    file_metadata[key] = value
+                else:
+                    file_comments.append(line)
+                continue
+
+            url = line
+            meta = ''
+            url_index = url_index + 1
+
+            # handle cases where URL is enclosed in quotes
+            if line[0] == '"':
```

Comment:
why are we even considering this as a case? Just use a csv parser from the csv library and it will automatically interpret quotes for you-> everything inside quotes is a single item, even if there are commas in the quote